Refactor the partitioning flow (#20)
intarchboard/draft-obliviousness: draft-kuepau-iab-privacy-partitioning.md

# Introduction

Protocols such as TLS and IPsec provide a secure (authenticated and encrypted) channel
between two endpoints over which endpoints transfer information. Encryption and authentication
of data in transit is necessary to protect information from being seen or modified by parties

> other than the intended protocol participants. As such, it is necessary for ensuring that

Reviewer: ```suggestion
other than the intended protocol participants. As such, this kind of security is necessary for ensuring that
```

Saying "it" didn't make sense, since we are referring to "encryption and authentication", I think

information transferred over these channels remain private.

However, a secure channel between two endpoints is insufficient for privacy of the endpoints

> themselves. In recent years, privacy has expanded beyond the need to protect data in transit

Reviewer: ```suggestion
themselves. In recent years, privacy requirements have expanded beyond the need to protect data in transit
```

such as a news article, without needing to have which specific articles they
read on their account being recorded. This is problematic for privacy since the service
can link user activity to the user's account.

- A client device that needs to upload metrics to an aggregation service might want to be
able to contribute data to the system without having their specific contributions being
attribued to them. This is problematic for privacy since the service can link client
contributions to the specific client.

The commonality in these examples is that clients want to interact with or use a service
without exposing too much user-specific or identifying information to that service. In particular,
separating the user-specific identity information from user-specific data is necessary for
privacy. Thus, order to protect user privacy, it is important to keep identity (who) and data
(what) separate.

This document defines "privacy partitioning" as the general technique used to separate the data
and metadata visible to various parties in network communication, with the aim of improving
user privacy. Partitioning is a spectrum and not a panacea. It is difficult to guarantee there

> is no link between user-specific identity and user-specific data. However, applied properly

Reviewer: ```suggestion
is no link between user-specific identity and user-specific data. However, applied properly,
```

privacy partitioning helps ensure that user privacy violations becomes more technically difficult
to achieve over time.

Several IETF working groups are working on protocols or systems that adhere to the principle
of privacy partitioning, including OHAI, MASQUE, Privacy Pass, and PPM. This document summarizes

> work in those groups and framework for reasoning about the resulting privacy posture of different

Reviewer: ```suggestion
work in those groups and describes a framework for reasoning about the resulting privacy posture of different
```

endpoints in practice.

# Privacy Partitioning

For the purposes of user privacy, this document focuses on user-specific information. This
might include any identifying information that is specific to a user, such as their email
address or IP address, or data about the user, such as their date of birth. Informally,
the goal of privacy partitioning is to ensure that each party in a system beyond the user
themselves only has access to one type of user-specific information.

This is a simple application of the principle of least privilege, wherein every party in
a system only has access to the minimum amount of information needed to fulfill their
function. Privacy partitioning advocates for this minimization by ensuring that protocols,
applications, and systems only reveal user-specific information to parties that need access
to the  information for their intended purpose.

Put simply, privacy partitioning aims to separate *who* someone is from *what* they do. In the

> rest of this section, we describe how the privacy partitioning can be used to achieve this goal.

Reviewer: ```suggestion
rest of this section, we describe how privacy partitioning can be used to achieve this goal.
```

## Privacy Contexts

Each piece of user-specific information exists within some context, where a context
is abstractly defined as a set of data and metadata and the entities that share access

> to that metadata. In order to prevent correlation of user-specific information across

Reviewer: ```suggestion
to that information. In order to prevent correlation of user-specific information across
```
Since this is both data and metadata...

~~~
{: #diagram-dualconnect title="Diagram of making separate connections to generate separate contexts"}

## Context Separation

> In order to define an analyze how various partitioning techniques work, the boundaries of what is

Reviewer: ```suggestion
In order to define and analyze how various partitioning techniques work, the boundaries of what is
```

Creating separate communication contexts is one important aspect of partioning, but
ensuring that the contexts are split to prevent correlation of identities is necessary
to make the partitions effective for improving privacy.

{{?RFC6973}} discusses the importance of identifiers in reducing correlation:

"Correlation is the combination of various pieces of information related to an individual
or that obtain that characteristic when combined... Correlation is closely related to
identification.  Internet protocols can facilitate correlation by allowing individuals'
activities to be tracked and combined over time."

{{?RFC6973}} goes on to discuss different ways that identities can be obscured: anonymity
and pseudonymity.

"To enable anonymity of an individual, there must exist a set of individuals that appear to
have the same attribute(s) as the individual."

"Pseudonymity is strengthened when less personal data can be linked to the pseudonym; when
the same pseudonym is used less often and across fewer contexts; and when independently
chosen pseudonyms are more frequently used for new actions (making them, from an observer's or
attacker's perspective, unlinkable)."

> The "anonymity level" of a given identity exists on a scale, not a clear line between real

Reviewer: I think it's still useful to have a subsection that mentions identity as seen within a context?

Author: I don't know if it's useful to zero in on identity since it's just one type of user-specific information. I think it's more helpful to just say that "each context has at most one type of user-specific information."

Author: (Discussed offline) Pushed back some text from RFC6973.
